flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CompletedCheckpointStore.java

```diff
@@ -45,8 +47,12 @@
      *
      * <p>Only a bounded number of checkpoints is kept. When exceeding the maximum number of
      * retained checkpoints, the oldest one will be discarded.
+     *
+     * @return the subsumed oldest completed checkpoint if possible, return null if no checkpoint
+     *     needs to be discarded on subsume.
      */
-    void addCheckpoint(
+    @Nullable
+    CompletedCheckpoint addCheckpointAndSubsumeOldestOne(
```

Comment:
The new name doesn't seem to reflect what the method actually does.
It doesn't necessarily subsume **one** checkpoint (or **any** checkpoints at all).

The correct name would be `addAndMaybeSubsumeOldestAndReturnTheLatestSubsumedIfAny` - which clearly indicates to me that the new functionality should be in a separate method.